File: InsertDeleteGetRandomO(1).py
import random

class RandomizedSet:

    def __init__(self):
        self.values = []
        self.map = {}

    def insert(self, val: int) -> bool:
        if val in self.map:
            return False
        self.map[val] = len(self.values)
        self.values.append(val)
        return True

    def remove(self, val: int) -> bool:
        if val in self.map:
            i = self.map.pop(val)
            self.values[i] = self.values[len(self.values)-1]
            self.values.pop()
            if i < len(self.values):
                self.map[self.values[i]] = i
            return True
        return False

    def getRandom(self) -> int:
        rand = random.randint(0, len(self.values)-1)
        return self.values[rand]
        


# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()

File: Arrays/InsertDeleteGetRandomO(1).cs
public class RandomizedSet {
    HashSet<int> randomSet;
    List<int> elements;
    Random random = new Random();

    public RandomizedSet() {
        randomSet = new HashSet<int>();
        elements = new List<int>();
    }
    
    public bool Insert(int val) {
        if(randomSet.Contains(val)){
            return false;
        }
        else{
            randomSet.Add(val);
            elements.Add(val);
            return true;
        }
    }
    
    public bool Remove(int val) {
        if(randomSet.Contains(val)){
            randomSet.Remove(val);

            int valIndex = elements.IndexOf(val);
            int lastElement = elements[elements.Count - 1];
            elements[valIndex] = lastElement;
            elements.RemoveAt(elements.Count - 1);

            return true;
        }
        else{
            return false;
        }
    }
    
    public int GetRandom() {
        int randomIndex = random.Next(elements.Count); // Generate a random index
        return elements[randomIndex];
    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet obj = new RandomizedSet();
 * bool param_1 = obj.Insert(val);
 * bool param_2 = obj.Remove(val);
 * int param_3 = obj.GetRandom();
 */
